compiler/rustc_codegen_llvm/src/gotoc/cbmc/goto_program/expr.rs

```diff
@@ -1205,6 +1226,12 @@ impl Expr {
         res.overflowed.ternary(saturating_val, res.result)
     }
 
+    /// `"s"`
+    /// only to be used when manually wrapped in `.array_to_ptr()`
```

Comment:
Can we add a constructor that enforces this invariant?

Reply:
Maybe I should make the comment more clear, but this means manually wrapped in `.array_to_ptr()` *after* construction (the other constructor does it for you).